jupyterhub/values.yaml

```diff
@@ -55,6 +55,10 @@ hub:
     requests:
       cpu: 200m
       memory: 512Mi
+  containerSecurityContext:
+    runAsUser: *hubUid
+    # Don't allow any process to execute as root inside the container
+    allowPrivilegeEscalation: false
```

Comment:
To be clear, leave this in the default values for `hub.containerSecurityContext` and if someone overrides `hub.containerSecurityContext` and wants to retain `allowPrivilegeEscalation: false` then they need to explicitly declare it in their override? I'm fine with that, but just want to make sure I understand the intent.

Reply:
Helm will merge dictionary values, not replace the entire dictionary that is available in the default. Arrays will be replaced though. Example:

```yaml
hub:
  containerSecurityContext:
    allowPrivilegeEscalation: true
    something-else: test
```

And the defaults are:

```yaml
hub:
  containerSecurityContext:
    runAsUser: 1000
    allowPrivilegeEscalation: false
```

Then the result would be:

```yaml
hub:
  containerSecurityContext:
    runAsUser: 1000
    allowPrivilegeEscalation: true
    something-else: test
```

Reply:
If you assumed it would be replaced entirely, perhaps your discussion in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1708#issuecomment-654319905 is influenced ?

Reply:
Hi again @mriedem, just making sure you saw this comments. I often miss them when they are not at the bottom of the PR

Reply:
Hey @consideRatio sorry yeah I didn't see this and yeah I didn't realize helm would merge those values. So given that, I _think_ this is ready to go? The `hub.uid` value is removed and there is a deprecation warning in NOTES.txt. We're still hard-coding the `runAsUser` value to 1000 like before so no changes downstream if people weren't modifying `hub.uid`, and `allowPrivilegeEscalation` stays the same.

So I think that means for my needs I can just have this in my values.yaml:

```yaml
hub:
  containerSecurityContext:
    capabilities:
      add:
      - SYS_PTRACE
```

And should end up with this:

```yaml
hub:
  containerSecurityContext:
    runAsUser: 1000
    allowPrivilegeEscalation: false
    capabilities:
      add:
      - SYS_PTRACE
```

If so, that's perfect.